java/src/jmri/InstanceManager.java

```diff
@@ -112,7 +126,7 @@
      * @param item The object of type T to be deregistered
      * @param type The class Object for the item's type.
      */
-    static public <T> void deregister(T item, Class<T> type) {
+    static public <T> void deregister(T item, @Nonnull Class<T> type) {
```

Comment:
Should item be @Nonnull here?
We currently log an error if a null object is registered, but appear to allow it anyway.